### PR TITLE
Add Hashicorp packer binary

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by dhirschfeld** [comment]
This is a repackaging of the [`packer.exe`](https://www.packer.io/) binary by Hashicorp.

The motivation is to allow me to use my package manager of choice (conda) to manage all my dev dependencies.

**Comment by dhirschfeld** [comment]
I think this is good to go, pending review.

I've placed the binary in `CONDA_ROOT\bin` - is this the correct location or should it be in `CONDA_ROOT\Library\bin`? I'm a bit confused about their relationship to each other 🤔

**Comment by mariusvniekerk** [comment]
Could we build packer instead?  We have quite a few of the hashicorp stuff building and packaged already

**Comment by dhirschfeld** [comment]
I'm just looking at `terraform` actually and unfortunately Windows is being skipped there:
https://github.com/conda-forge/terraform-feedstock/blob/master/recipe/meta.yaml#L20

...because no one can build a modern enough go:
https://github.com/conda-forge/go-feedstock/issues/63

I'm not sure what the minimum go version is required to build packer but I worry that we may run into the same blocker on Windows.

**Comment by dhirschfeld** [comment]
I agree in an ideal world that we should build from source for all platforms but, even if I were capable of figuring out how to do so I won't have time in the immediate future.

Since this only installs the binary on Windows maybe we could allow it for now and open an issue to build from source?

**Comment by dhirschfeld** [comment]
^^^^Nice! 🤞

File: recipes/packer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{% set goname = "github.com/hashicorp/packer" %}
+{% set version = "0.12.18" %}
```
Comment (dhirschfeld) [inline recipes/packer/meta.yaml]:
```suggestion
{% set version = "1.5.1" %}
```

File: recipes/packer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  # We must use git here beause the program looks
+  # for the git sha&tag to determine its version
+  - folder: {{ pkg_src }}
+    url: https://{{ goname }}/archive/v{{ version }}.tar.gz
+    sha256: 1253a870ba2c1d9a873f6bfcaa2e4450573fd06177705f0faeb451157e1dde8b
```
Comment (dhirschfeld) [inline recipes/packer/meta.yaml]:
```suggestion
    sha256: 97ba9fd9615259cfab3601daa81d81bba88ac1107633dc7f83e3ada06306fc5f
```

**Comment by dhirschfeld** [comment]
Sorry, have to leave it for now but will check back after xmas!

Thanks for taking a look & I hope you have a merry xmas! 🎅 🍰 🎁 🎄

**Comment by mariusvniekerk** [comment]
I dont think we have a `go` compiler for windows yet

**Comment by mariusvniekerk** [comment]
Reverting all the staged recipes hacks now and removed the binary repack, since we can get a proper build out now.

**Pull request merged**